lua/lazyvim/plugins/extras/lang/typst.lua

```diff
@@ -0,0 +1,62 @@
+return {
+  recommended = function()
+    return LazyVim.extras.wants({
+      ft = { "typst" },
+    })
+  end,
+
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = {
+      ensure_installed = { "typst" },
+    },
+  },
+
+  {
+    "neovim/nvim-lspconfig",
+    opts = {
+      servers = {
+        tinymist = {
+          single_file_support = true, -- Fixes LSP attachment in non-Git directories
+          root_dir = LazyVim.root.get,
+          settings = {
+            formatterMode = "typstyle",
+          },
+        },
+      },
+    },
+  },
+
+  {
+    "stevearc/conform.nvim",
+    optional = true,
+    opts = {
+      formatters_by_ft = {
+        typst = { "typstyle", lsp_format = "prefer" },
+      },
+    },
+  },
+
+  {
+    "chomosuke/typst-preview.nvim",
+    cmd = { "TypstPreview", "TypstPreviewToggle", "TypstPreviewUpdate" },
+    build = function(plugin)
+      local typst_preview = require("typst-preview")
+      typst_preview.setup(plugin.opts) -- Setup preview during build to avoid redundant downloads
```

Comment:
remove this line

Reply:
Removing this line may cause the plugin to redundantly download the tinymist binary, even though we're already using tinymist via Mason. Any suggestions to prevent this?

Reply:
```lua
      local typst_preview = require("typst-preview") -- this will trigger lazy to setup the plugin
      --- typst_preview.setup(plugin.opts) -- remove just this line
```

Reply:
I tested, it still download `tinymist` binary.

Reply:
Sorry, I don't think I understand what you want to achieve here then. Can you please explain?

Reply:
The typst-preview plugin requires both tinymist and websocat binaries. Since we're already using the tinymist binary downloaded by mason, we don’t need typst-preview to download it again (on Linux, we can check at `~/.local/share/nvim/typst-preview`). By setting `dependencies_bin.tinymist` to a non-nil value, we can prevent this redundant download. That's the goal here.

Reply:
The `typst_preview.update()` will download the tinymist and websocat

Reply:
Yes

Reply:
No it does not, or yes it does?

Reply:
Just `require("typst-preview").update()` in `build` does not achieve what I want

Reply:
but why does this need to be ran during build? This is all so werid.
Again, what you have there won't always work. plugin.opts is not the (always) the table that will get passed to setup, so dont use that!

Reply:
What if you do
```lua
local typst_preview = require("typst-preview")
local fetch = require("typst-preview.fetch")
-- Set custom binary names
fetch.tinymist_bin_name = "tinymist"
typst_preview.update()
```
Would that work? It seems that [here](https://github.com/chomosuke/typst-preview.nvim/blob/master/lua/typst-preview/fetch.lua) is where the `update()` happens and it essentially calls `fetch()`. And the functions to get the bin's name check the top table 
```lua
local M = {
  -- Exposing this so when platform detection fails user can manually set
  -- bin_name
  tinymist_bin_name = nil,
  websocat_bin_name = nil,
}
```
to see if to get the bin's name or not.

Reply:
From my testing, it looks like the opts are from the spec are not merged with the default opts from the plugin before/during build when build is first called during install - despite the `require("typst-preview")`. I'm not sure why that is. I managed to fix it by manually setting the required option with
```lua
require("typst-preview.config").opts.dependencies_bin["tinymist"] = "tinymist"
```
rather than calling the plugin's setup in `build`

Reply:
But the problem with this approach might be that if a user wants to use the downloaded binary from the plugin, he would also have to change the `build` function.

Reply:
@folke as far as I can tell, a plugin's setup is not called during the `build` function when the plugin is first installed, even if it is `require`d. This is not the case after the first install, and the plugin is built with eg `:Lazy build <plugin>`. Is this intended behavior in `lazy.nvim`?

Reply:
yes, that's actually on purpose. Forgot about that. Just the path is added to the rtp, but setup is not ran, since most plugins need their build to be completed before doing setup.

Why does this need to be part of build anyway? Won't the plugin download it when needed?

Reply:
Just checked, and the plugin **will not** download when needed, only during update. When toggling preview without a build function, the plugin prints this error:

![image](https://github.com/user-attachments/assets/1a5e8084-7cb9-4343-8b9f-1ff74d7ed7f5)

I suppose the simplest solution would be to leave out a `build` and submit a FR to the plugin to download the executables as needed, and for now, whenever a user encounters the error they can run the command.

Reply:
And why not download/install both during build? IS the download that big?

Reply:
The latest `tinymist` is 52MB, which is not too large but does take a bit (for me around 10s). I guess leaving the build as `require('typst-preview").update()` and having it download the binary a second time isn't too big of a deal, just slightly annoying. Also, this is only an issue when the plugin is installed, so afterwards the binary can safely be removed and it won't be downloaded again (because for subsequent builds, the plugin is configured).

Reply:
> What if you do
> 
> ```lua
> local typst_preview = require("typst-preview")
> local fetch = require("typst-preview.fetch")
> -- Set custom binary names
> fetch.tinymist_bin_name = "tinymist"
> typst_preview.update()
> ```
> 
> Would that work? It seems that [here](https://github.com/chomosuke/typst-preview.nvim/blob/master/lua/typst-preview/fetch.lua) is where the `update()` happens and it essentially calls `fetch()`. And the functions to get the bin's name check the top table
> 
> ```lua
> local M = {
>   -- Exposing this so when platform detection fails user can manually set
>   -- bin_name
>   tinymist_bin_name = nil,
>   websocat_bin_name = nil,
> }
> ```
> 
> to see if to get the bin's name or not.

Sorry, I missed this earlier. I don't believe this will work because `tinymist_bin_name` is the binary to download from github, but is not used if `opts.dependencies_bin.tinymist` is set, which is checked [here](https://github.com/chomosuke/typst-preview.nvim/blob/06778d1b3d4d29c34f1faf80947b586f403689ba/lua/typst-preview/fetch.lua#L107)

Reply:
I tested and it doesn't download `tinymist`. It works because changing `require("typst-preview.fetch").tinymist_bin_name` it also changes the url of the bin in `M.bins_to_fetch()`, so it can't find the binary from the wrong url.